make/autoconf/flags-cflags.m4

```diff
@@ -758,8 +692,7 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_CPU_DEP],
   $1_DEFINES_CPU_JDK="${$1_DEFINES_CPU_JDK} -DARCH='\"$FLAGS_CPU_LEGACY\"' \
       -D$FLAGS_CPU_LEGACY"
 
-  if test "x$FLAGS_CPU_BITS" = x64 && test "x$FLAGS_OS" != xaix; then
-    # xlc on AIX defines _LP64=1 by default and issues a warning if we redefine it.
+  if test "x$FLAGS_CPU_BITS" = x64; then
```

Comment:
A wise man said: "If the code and the comments contradict each other, then probably both are wrong".

I am here assuming that the comment claiming that this is only needed by xlc is correct. If it turns out that this is needed even with clang on AIX, we'll have to restore the test and update the comment to this fact.